### DISCUSSION TITLE
Is ticket renewal working as designed?

### DISCUSSION THREAD

**Discussion by IFYates** [dotnet/aspnetcore#27278] [General]
Hi. I'm not raising this as an issue incase I've misunderstood, but I wonder if current expiration logic is as expected.

[This article](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.cookies.cookieauthenticationoptions.slidingexpiration?view=aspnetcore-3.1) states that the cookie is re-issued:

> with a new expiration time any time it processes a request which is more than halfway through the expiration window

It looks like this is achieved through setting `_shouldRenew` to `true`: https://github.com/dotnet/aspnetcore/blob/4fdc35d058dd86398ec2989dfc2fe25f7a95f0a4/src/Security/Authentication/Cookies/src/CookieAuthenticationHandler.cs#L108
However, I can't see that this value ever goes `false` again, so every subsequent request causes a call to `RenewAsync`.

In my tests, setting the `ExpireTimeSpan` of the `CookieAuthenticationOptions` instance to (e.g.) 10 minutes correctly exposes the cookie on the creation response, and then not again until 5 minutes later. However, every response after that first 5 minutes includes the cookie with an updated expiry time.

It's not causing us any issues, but it seemed odd so thought I'd reach out.

Cheers

**Comment by Tratcher** [comment]
CookieAuthenticationHandler should be re-created per request so that should be resetting the _shouldRefresh field. If you're somehow re-using the CookieAuthenticationHandler across requests that would be bad.

**Reply by IFYates** [reply]
Thanks. That makes sense. That explains another issue we're seeing, all related to the token issue time being incorrect.
